service/search/githubsearch.go

```diff
@@ -0,0 +1,68 @@
+package search
+
+import (
+	"context"
+	"fmt"
+	"githubsearchservice/pkg"
+
+	"github.com/go-resty/resty/v2"
+)
+
+type GithubSearcher struct {
+	githubAPIClient *resty.Client
+}
+
+func NewGithubSearcher(githubAPIClient *resty.Client) *GithubSearcher {
+	return &GithubSearcher{
+		githubAPIClient: githubAPIClient,
+	}
+}
+
+type repository struct {
+	FullName string `json:"full_name"`
+}
+
+type item struct {
+	HTMLURL    string     `json:"html_url"`
+	Repository repository `json:"repository"`
+}
+
+type githubSearchAPIResponse struct {
+	Items []item `json:"items"`
+}
+
+func (gs *GithubSearcher) Search(ctx context.Context, query string) ([]Result, error) {
+	var searchResp githubSearchAPIResponse
+
+	githubAPIToken := pkg.GetGithubAPIToken(ctx)
+
+	resp, err := gs.githubAPIClient.R().
+		SetHeader("Authorization",
+			fmt.Sprintf("token %v", githubAPIToken)).
+		SetContext(ctx).
+		SetQueryParam("q", query).
+		SetResult(&searchResp).
+		Get("search/code")
+
```

Comment:
Small suggestion for readability.

```suggestion
	var searchResp githubSearchAPIResponse
	resp, err := gs.githubAPIClient.R().
		SetHeader("Authorization",
			fmt.Sprintf("token %v", pkg.GetGithubAPIToken(ctx))).
		SetContext(ctx).
		SetQueryParam("q", query).
		SetResult(&searchResp).
		Get("search/code")

```